### PR TITLE
refactor: add theme provider to toggle theme without app reload

### LINKED ISSUE #48: Theme change should not require page reload
Right now changing theme from light to dark requires the page to reload. This is not necessary. The value should be stored in a Theme provider which also exposes a switch between `light` and `dark` theme and is responsible for writing it into local storage.

### PR DESCRIPTION
**Pull request by vojtechsimetka**
Resolves #48

### CODE REVIEW

**Review by matmertz25 — APPROVED** [review]
Thanks for adding this, works much better than a reload

**Comment by vojtechsimetka** [comment]
Does it work for you? In my case it does not switch well for the second time. No idea why but I think we'll find out why after some more refactoring :)

**Comment by matmertz25** [comment]
Yeah your right, it initially works for me but after flipping back and forth per your comment it is not transitioning from dark -> light correctly.

File: src/providers/Theme.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        localStorage.setItem('theme', newThemeMode)
+    }
+
+    // Detect previously store theme
+    useEffect(() => {
+        const newThemeMode: ThemeMode = localStorage.getItem('theme') === 'light' ? 'light' : 'dark'
+
+        setThemeMode(newThemeMode)
+
+        window?.matchMedia('(prefers-color-scheme: dark)')?.addEventListener('change', e => {
```
Comment (agazso) [inline src/providers/Theme.tsx:38]: You pass an anonymous function to the `addEventListener` and another (but functionally same) anonymous function to `removeEventListener`. This is undefined (browser-dependent) behavior and usually it's better to extract the anonymous function and assign it to a variable, and pass the variable to the `addEventListener` and `removeEventListener` calls.

**Comment by vojtechsimetka** [comment]
This should be looked at with fresh eyes. We probably don't need and want the provider altogether (as it is likely the reason why it does not work well) and get inspired by the official documentation https://material-ui.com/customization/palette/#user-preference

**Pull request closed without merging**